src/main/kotlin/lotto/domain/Money.kt

```diff
@@ -0,0 +1,40 @@
+package lotto.domain
+
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+@JvmInline
+value class Money(
+    private val value: BigDecimal,
+) : Comparable<Money> {
+    init {
+        require(value in MINIMUM_MONEY..MAXIMUM_MONEY) {
+            "Amount value must be between 0 and 100,000"
+        }
+    }
+
+    constructor(value: Int) : this(BigDecimal(value))
+
+    fun toInt() = value.toInt()
```

Comment:
This seems like a lose-lose situation of wrapped value, isn't it? 🤧

Using DTOs might help, but that's another trade-off. But it's worth thinking about how much the view needs to know about the domain model.

Reply:
I slacked you for further explanation 🙌